index/server/main.go

```diff
@@ -36,19 +37,22 @@ var (
 
 func main() {
 	// Wait until registry is up and running
-	isDone := false
-	for !isDone {
+	err := wait.PollImmediate(time.Millisecond, time.Second*30, func() (bool, error) {
 		resp, err := http.Get(scheme + "://" + registryService)
 		if err != nil {
-			log.Fatal(err.Error())
+			return false, err
```

Comment:
I'm still hitting the crash on first time start up:
```
Johns-MacBook-Pro-3:registry-operator johncollier$ k logs -f -c devfile-registry-bootstrap test-devfile-registry-cffb7559b-nbspk
2020/12/01 17:59:45 Get "http://localhost:5000": dial tcp 127.0.0.1:5000: connect: connection refused
/entrypoint.sh: fg: line 19: job %1 not created under job control
```

Looks like this was an issue with the sample I provided you. It looks like `wait.PollImmediate` will return an error early if the function passed into it ever returns a non-nil error.

So I think we need to change this line to: `return false, nil`

Reply:
Yup, confirmed that was the issue. I would change the `if err != nil` to:
```
if err != nil {
			log.Println(err.Error())
			return false, nil
		}
```

Log the error too so that it's not lost. We just don't want to return it.

Reply:
My original understanding was that only return `true, err` can cause the polling function throw error within timeout, but seems like from your testing it doesn't work like that. But interesting thing is my cluster doesn't throw the error even with return `false, err` (probably it depends on the time of registry is up and running).

My concern is that if we return `false, nil` and we don't return it, then the real registry failure won't bring the server down.

Reply:
Well, until `true, nil` is returned, the polling function will keep retrying up until the timeout is reached. So if there is something wrong, the timeout will be reached and an error will be returned.

Reply:
I check the code again, I that would be fine as we can bring the server down on pushing devfile stage instead of checking registry is up or not stage, not perfect, but still works fine, only delay couple of seconds I think.

Reply:
I am gonna make the changes as you suggest.